src/libsync/discoveryphase.cpp

```diff
@@ -704,16 +677,14 @@ csync_vio_handle_t *DiscoveryJob::remote_vio_opendir_hook(const char *url,
 }
 
 
-csync_vio_file_stat_t *DiscoveryJob::remote_vio_readdir_hook(csync_vio_handle_t *dhandle,
+std::unique_ptr<csync_file_stat_t> DiscoveryJob::remote_vio_readdir_hook(csync_vio_handle_t *dhandle,
     void *userdata)
 {
     DiscoveryJob *discoveryJob = static_cast<DiscoveryJob *>(userdata);
     if (discoveryJob) {
         DiscoveryDirectoryResult *directoryResult = static_cast<DiscoveryDirectoryResult *>(dhandle);
         if (directoryResult->listIndex < directoryResult->list.size()) {
-            csync_vio_file_stat_t *file_stat = directoryResult->list.at(directoryResult->listIndex++).data();
-            // Make a copy, csync_update will delete the copy
-            return csync_vio_file_stat_copy(file_stat);
+            return std::move(directoryResult->list.at(directoryResult->listIndex++));
```

Comment:
You are moving from an item in the list?  leaving a moved-from item in the list?
That's OK. But maybe having a queue data structure would be better.

Reply:
Humm but then the index won't make sense. Since the caller just calls this repetitively without providing the index, this should be OK, no?

Reply:
Although yeah I could just remove that weird index that's true.

Reply:
Done, used an `std::deque`